test/powershell/Modules/Microsoft.PowerShell.Management/Set-Service.Tests.ps1

```diff
@@ -329,4 +329,21 @@ Describe "Set/New/Remove-Service cmdlet tests" -Tags "Feature", "RequireAdminOnW
         }
         { & $cmdlet @parameters } | Should -Throw -ErrorId $errorid
     }
+
+    It "Set-Service can run -Status Stopped to stop a service with dependencies" {
+        $servicename = "spooler"
+        $service = Get-Service -Name $servicename -ErrorAction SilentlyContinue
```

Comment:
I think this code should look more like:

```powershell
$service = Get-Service -Name $servicename
$service | Should -Not -BeNullOrEmpty
$originalStatus = $service.Status
try {
  Set-Service -Status Running $servicename
  (Get-Service $servicename).Status | Should -BeExactly "Running"
  Set-Service -Status Stopped $servicename
  (Get-Service $servicename).Status | Should -BeExactly "Stopped"
}
finally {
  Set-Service -Status $originalStatus $servicename
}
```

In general, you should not use `-ErrorAction SilentlyContinue` in the tests unless a failure is acceptable and doesn't impact the test result.